tests/src/test/scala/actionContainers/PythonActionContainerTests.scala

```diff
@@ -140,6 +141,76 @@ class PythonActionContainerTests extends BasicActionRunnerTests with WskActorSys
         })
     }
 
+    it should "run zipped Python action containing a virtual environment" in {
+        val zippedPythonActionWrongName = TestUtils.getTestActionFilename("action_python_virtualenv.zip")
+
+        val code = readAsBase64(Paths.get(zippedPythonActionWrongName))
+
+        val (out, err) = withActionContainer() { c =>
+            val (initCode, initRes) = c.init(initPayload(code, main = "main"))
+            initCode should be(200)
+            val args = JsObject("msg" -> JsString("any"))
+            val (runCode, runRes) = c.run(runPayload(args))
+            runCode should be(200)
+            runRes.get.toString() should include("netmask")
+        }
+        checkStreams(out, err, {
+            case (o, e) =>
+                o should include("netmask")
+                e shouldBe empty
+        })
+    }
+
+    it should "report error if zipped Python action has wrong main module name" in {
+        val zippedPythonActionWrongName = TestUtils.getTestActionFilename("action_python_virtualenv_name.zip")
+
+        val code = readAsBase64(Paths.get(zippedPythonActionWrongName))
+
+        val (out, err) = withActionContainer() { c =>
+            val (initCode, initRes) = c.init(initPayload(code, main = "main"))
+            initCode should be(502)
+        }
+        checkStreams(out, err, {
+            case (o, e) =>
+                o shouldBe empty
+                e should include("Zip file does not include __main__.py")
+        })
+    }
+
+    it should "report error if zipped Python action has invalid virtualenv directory" in {
+        val zippedPythonActionWrongDir = TestUtils.getTestActionFilename("action_python_virtualenv_dir.zip")
+
+        val code = readAsBase64(Paths.get(zippedPythonActionWrongDir))
+        val (out, err) = withActionContainer() { c =>
+            val (initCode, initRes) = c.init(initPayload(code, main = "main"))
+            initCode should be(502)
+        }
+        checkStreams(out, err, {
+            case (o, e) =>
+                o shouldBe empty
+                e should include("Zip file does not include /virtualenv/bin/")
+        })
+    }
+
+    it should "report error if zipped Python action has virtualenv created in wrong environment" in {
+        val zippedPythonActionWrongEnvironment = TestUtils.getTestActionFilename("action_python_virtualenv_env.zip")
+
+        val code = readAsBase64(Paths.get(zippedPythonActionWrongEnvironment))
+
+        val (out, err) = withActionContainer() { c =>
+            val (initCode, initRes) = c.init(initPayload(code, main = "main"))
+            initCode should be(200)
+            val args = JsObject("msg" -> JsString("any"))
+            val (runCode, runRes) = c.run(runPayload(args))
+            runCode should be(502)
+        }
+        checkStreams(out, err, {
+            case (o, e) =>
+                o shouldBe empty
+                e should include("Exec format error")
```

Comment:
is this an error you can easily catch and rewrite to a more elaborate error message?